crates/vm/levm/src/precompiles.rs

```diff
@@ -95,12 +98,70 @@ pub fn execute_precompile(current_call_frame: &mut CallFrame) -> Result<Bytes, V
     Ok(result)
 }
 
-fn ecrecover(
-    _calldata: &Bytes,
-    _gas_for_call: U256,
-    _consumed_gas: &mut U256,
+fn fill_with_zeros(slice: &[u8]) -> [u8; 128] {
+    let mut result = [0; 128];
+
+    let n = slice.len().min(128);
+
+    let trimmed_slice = slice.get(..n).unwrap_or_default();
+    result
+        .get_mut(..n)
+        .unwrap_or_default()
+        .copy_from_slice(trimmed_slice);
+
+    result
+}
+
+pub fn ecrecover(
+    calldata: &Bytes,
+    gas_for_call: U256,
+    consumed_gas: &mut U256,
 ) -> Result<Bytes, VMError> {
-    Ok(Bytes::new())
+    // Consume gas
+    *consumed_gas = consumed_gas
+        .checked_add(ECRECOVER_COST.into())
+        .ok_or(PrecompileError::GasConsumedOverflow)?;
+
+    if gas_for_call < *consumed_gas {
+        return Err(VMError::PrecompileError(PrecompileError::NotEnoughGas));
+    }
+
+    // If calldata does not reach the required length, we should fill the rest with zeros
+    let calldata = fill_with_zeros(calldata);
+
+    let hash = calldata
+        .get(0..32)
+        .ok_or(PrecompileError::ParsingInputError)?;
+    let message = Message::parse_slice(hash).map_err(|_| PrecompileError::ParsingInputError)?;
+
+    let r = calldata.get(63).ok_or(PrecompileError::ParsingInputError)?;
+    let recovery_id = match RecoveryId::parse_rpc(*r) {
+        Ok(id) => id,
+        Err(_) => {
+            return Ok(Bytes::new());
```

Comment:
Is this expected behavior?

Reply:
I don't really know, but makes more tests pass. I think this proves that there's a case we are not handling in the `VM::execute`. The same happens in the match you've pointed below